chain/chain/src/flat_storage_resharder.rs

```diff
@@ -720,17 +747,16 @@ impl FlatStorageResharder {
             }
         }
         // Set the flat storage status to `Ready`.
-        store_update.set_flat_storage_status(
-            shard_uid,
-            FlatStorageStatus::Ready(FlatStorageReadyStatus {
-                flat_head: BlockInfo {
-                    hash: flat_head.last_block_hash,
-                    prev_hash: flat_head.prev_block_hash,
-                    height: flat_head.height,
-                },
-            }),
-        );
+        let flat_storage_status = FlatStorageStatus::Ready(FlatStorageReadyStatus {
+            flat_head: BlockInfo {
+                hash: flat_head.last_block_hash,
+                prev_hash: flat_head.prev_block_hash,
+                height: flat_head.height,
+            },
+        });
+        store_update.set_flat_storage_status(shard_uid, flat_storage_status.clone());
```

Comment:
ditto - isn't it already done for flat storage?

Reply:
`create_flat_storage_for_shard ` expects flat storage to be explicitly set to `Ready`, or did you mean something else?

Reply:
I was asking about if the state metric isn't also set elsewhere already but you've answered on the other comment.